WikiExtension/src/timeSeriesService.js

```diff
@@ -25,24 +26,97 @@ const fetchPageViews = async (title, startDate, endDate, aggregateType) => {
+ * use if there are any older revisions. If older versions of revision exists, we will get
+ * {
+ *  revisions: [...],
+ *  ...
+ *  older: 'https://en.wikipedia.org/w/rest.php/v1/page/Jupiter/history?older_than=1103899458',
+ *  ...
+ * }
+ *
+ * Thus, we keep calling the endpoint with the given query if the older property exists in the
+ * previous request.
+ *
+ * @param {string} title of a wikipedia article
+ * @param {Date} startDate
+ * @param {Date} endDate
+ * @returns An array containing all revision objects
+ */
+const fetchPageRevisions = async (title, startDate, endDate) => {
+    const response = await fetch(`https://en.wikipedia.org/w/rest.php/v1/page/${title}/history`);
+    let curJsonResponse = await response.json();
+
+    let revisions = curJsonResponse.revisions;
+
+    while (curJsonResponse.hasOwnProperty("older")) {
+        const localResponse = await fetch(curJsonResponse.older);
+        curJsonResponse = await localResponse.json();
+
+        const localRevisions = curJsonResponse.revisions;
+
+        // Stop making requests when requesting passes startDate
+        const localLatestTimestamp = new Date(localRevisions[0].timestamp);
+        if (localLatestTimestamp < startDate) break;
+
+        revisions.push(...localRevisions);
+    }
+
+    // Filters revisions that are in between the given timerange
+    return revisions.filter((revisionObject) => {
+        const currentTimestamp = new Date(revisionObject.timestamp);
+        return startDate <= currentTimestamp && currentTimestamp <= endDate;
+    });
+};
+
 /**
  * TODO: Subject to change based on what the graph needs
- * @return formated fetchedPageViews as a list of lists that only contain date object and count
+ * @return formated fetchedPageViews as an array of arrays that only contain date object and count
  */
 const formatPageViews = (fetchedPageViews) => {
     return Array.from(fetchedPageViews.items).map((pageViewObject) => {
         return [formatYYYYMMDDToDate(pageViewObject.timestamp), pageViewObject.views];
     });
 };
 
+/**
+ * TODO: Subject to change based on what the graph needs
+ * @param {AggregateType} aggregateType
+ * @return formated fetchedRevisions as an array of arrays that only contain date object and count
+ */
+const formatPageRevisions = (fetchedRevisions, aggregateType) => {
+    const pageRevisionCountMap = fetchedRevisions
+        .map((revisionObject) => {
+            const date = new Date(revisionObject.timestamp);
+            date.setHours(0, 0, 0, 0); // have date only contains year, month, and day (not time)
+
+            // Set date to 1 to aggregate monthly instead of daily
+            if (aggregateType == AggregateType.MONTHLY) {
+                date.setDate(1);
+            }
+
+            return date;
+        })
+        .reduce((accumulator, timestamp) => {
+            accumulator[timestamp] = (accumulator[timestamp] ?? 0) + 1;
+            return accumulator;
+        }, {});
```

Comment:
reduces from something like 
```
[
  {
    id: 1065915668,
    timestamp: '2022-01-15T23:23:24Z',
    minor: false,
    size: 60364,
    comment: '/* See also */ +{{portal|Food}}',
    user: { id: 14730724, name: 'Northamerica1000' },
    delta: 15
  },
  {
    id: 1064467547,
    timestamp: '2022-01-08T14:23:37Z',
    minor: true,
    size: 60349,
    comment: '/* Production and market */Cleanup and [[WP:AWB/T|typo fixing]], [[WP:AWB/T|typo(s) fixed]]: 2015-16 → 2015–16',
    user: { id: 1273126, name: 'Iridescent 2' },
    delta: 2
  }
]
```
to 
```
{
  'Sat Jan 15 2022 00:00:00 GMT-0600 (Central Standard Time)': 1,
  'Sat Jan 08 2022 00:00:00 GMT-0600 (Central Standard Time)': 1
}
```

It goes through all the elements in the array and count elements that have the same timestamp (hence why we needed to set hours to 0).

Reply:
That makes sense, thanks!